Assets/Scripts/Player/PlayerCamera.cs

```diff
@@ -59,114 +82,160 @@ void Update()
 		}
 	}
 
-	void ToggleUnderWaterMode(bool setOn)
-	{
-		RenderSettings.fog = setOn;
-	}
-
-	void UnderWaterTest()
+	void EnterCameraState( PhysicsStateType physicsStateType )
 	{
-		RaycastHit hit = WadeUtils.RaycastAndGetInfo(cam.transform.position, Vector3.up, waterLayer, 10f);
-		if(hit.transform)
+		if( physicsStateType == PhysicsStateType.Gliding )
 		{
-			if(!underWater)
-			{
-				underWater = true;
-				ToggleUnderWaterMode(true);
-			}
+			_horRotationSpeed = _glideHorRotationSpeed;
+			_zoomDistance = _glideZoomDistance;
 		}
-		else if(underWater)
+		else
 		{
-			underWater = false;
-			ToggleUnderWaterMode(false);
+			_horRotationSpeed = _defaultHorRotationSpeed;
+			_zoomDistance = _defaultZoomDistance;
 		}
 	}
 
 	void CameraControl()
 	{
-		if ( cam )
+		// X - Camera movement is smooth
+
+		// X - Camera rotates on X and Y around focus point
+			// Y axis
+				// Camera can rotate freely around Y axis
+				// Rotation around axis is mouseY * yMod
+				// Rotation gets harder as you get close to the limit
+			// X axis
+				// Camera has limits on it's rotation
+				// Limits are up and down values
+				// Rotation around axis is mouseX * xMod
+
+		// X - Camera tries to maintain line of sight w/ the focus point
+			// if not player controlled, Camera will rotate around obstacles to keep view of its focus
+				// Cast rays to find the easiest path
+				// Worst case: cast a ray from the player towards our focus and zoom to the closest hit
+
+		// X - Camera tries to maintain offset from focus point
+
+		// Camera zooms to try and keep interesting elements on screen
+			// For interesting elements near the player, zoom out to show more
+
+		// Different modes of viewing:
+			// During platforming, we want a far out view
+			// While running, camera zooms in
+			// When dealing with buddies, we want a close up side-view
+			// Toggle between zoom/not zoom mode w/ R3
```

Comment:
Mouse scroll or middle click is what it would be. These comments aren't what the system is, just what I want it to be eventually